src/modules/sub-items/components/pagination/pagination.component.js

```diff
@@ -0,0 +1,126 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import PaginationBtnComponent from './pagination.btn.component';
+
+export default class PaginationComponent extends PureComponent {
+    getPagesNumber() {
+        const { totalCount, itemsPerPage } = this.props;
+
+        return Math.ceil(totalCount / itemsPerPage);
+    }
+
+    /**
+     * Computes array with pagination pages.
+     *
+     * Example 1: [ 1, "...", 5, 6, 7, 8, 9, 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 7)
+     * Example 2: [ 1, "...", 3, 4, 5, 6, 7, "...", 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 5)
+     * Example 3: [ 1, "...", 8, 9, 10, 11, 12, "...", 20 ] (for: proximity = 2; pagesNumber = 20; currentPage = 10)
+     *
+     * @returns {Array}
+     */
+    computePages() {
+        const { proximity, currentPage } = this.props;
+        const pagesNumber = this.getPagesNumber();
+        const pages = [];
+        let wereDots = false;
+
+        for (let i = 1; i <= pagesNumber; i++) {
+            const isFirstPage = i === 1;
+            const isLastPage = i === pagesNumber;
+            const isInRange = i >= (currentPage + 1) - proximity && i <= (currentPage + 1) + proximity;
+
+            if (isFirstPage || isLastPage || isInRange) {
+                pages.push(i);
+                wereDots = false;
+            } else if (!wereDots) {
+                pages.push('...');
+                wereDots = true;
+            }
+        }
+
+        return pages;
+    }
+
+    renderPrevious() {
+        const { currentPage, onPageChange } = this.props;
+        const previousLabel = Translator.trans(/*@Desc("Back")*/ 'pagination.previous', {}, 'sub_items');
+        const previousPage = currentPage - 1;
+        const isFirstPage = currentPage === 0;
+
+        return (
+            <PaginationBtnComponent
+                page={previousPage}
+                label={previousLabel}
+                additionalClasses="prev"
+                disabled={isFirstPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderNext() {
+        const { currentPage, onPageChange } = this.props;
```

Comment:
I would rename `currentPage` to `currentPageIndex`

Reply:
Mhm, I like this idea, you want to add `Index` to suggest that numeration in `currentPage` variable starts from 0? :)

Reply:
That would make sense. And it should be `activePage`, right?

Reply:
Right 👍

Reply:
I changed this but it might be a little bit inconsistent to have same value under `currentPage` and `currentPageIndex.`. 🤔  Nonetheless, I will keep it as it is.